packages/amplify-provider-awscloudformation/src/graphql-transformer/amplify-graphql-resource-manager.ts

```diff
@@ -103,6 +107,7 @@ export class GraphQLResourceManager {
       }
     }
     this.gsiManagement(gqlDiff.diff, gqlDiff.current, gqlDiff.next);
+    this.tableRecreationManagement(gqlDiff.current, gqlDiff.next);
```

Comment:
I think this should use the `rebuildAllTables` value since we don't need to check for gsi changes if we are rebuilding the tables anyway

Reply:
you mean wrap the `gsiManagement` method call in an `if (!this.rebuildAllTables)` check?

Reply:
Yea it's a different type of behavior for the whole api, I suppose down the line we can have a whole api type of resource check which manages the state of the whole api.

Reply:
though I guess implementation wise this is more of a nit.

Reply:
that's a valid optimization. I can make that change